### ISSUE TITLE
`no-immediate-mutation`: option to allow big values

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2816]
### Description

Maybe sometimes an assignment should be preferred due to the code size.

For example

```js
const object = {}
object.foo = function () {
  // the body only needs to indent one level
}
```

```js
const object = {
  foo() {
    // now requires 2-level indentation
  }
}
```

A real world case https://github.com/prettier/prettier/commit/9047a40a943a4896d0afd5fcd970a249ada1e562

**Comment by sindresorhus** [comment]
Going to pass on this.
